src/pair_potential_types.F

```diff
@@ -1795,6 +1830,85 @@ SUBROUTINE pair_potential_nequip_release(nequip)
+
+! **************************************************************************************************
+!> \brief Copy two ALLEGRO potential type
+!> \param allegro_source ...
+!> \param allegro_dest ...
+!> \author Teodoro Laino [teo] 11.2005
+! **************************************************************************************************
+   SUBROUTINE pair_potential_allegro_copy(allegro_source, allegro_dest)
+      TYPE(allegro_pot_type), POINTER                    :: allegro_source, allegro_dest
+
+      IF (.NOT. ASSOCIATED(allegro_source)) RETURN
+      IF (ASSOCIATED(allegro_dest)) CALL pair_potential_allegro_release(allegro_dest)
+      CALL pair_potential_allegro_create(allegro_dest)
+      allegro_dest%allegro_file_name = allegro_source%allegro_file_name
+      allegro_dest%allegro_version = allegro_source%allegro_version
+      allegro_dest%rcutsq = allegro_source%rcutsq
+      allegro_dest%unit_coords = allegro_source%unit_coords
+      allegro_dest%unit_coords_val = allegro_source%unit_coords_val
+      allegro_dest%unit_forces = allegro_source%unit_forces
+      allegro_dest%unit_forces_val = allegro_source%unit_forces_val
+      allegro_dest%unit_energy = allegro_source%unit_energy
+      allegro_dest%unit_energy_val = allegro_source%unit_energy_val
+      allegro_dest%unit_cell = allegro_source%unit_cell
+      allegro_dest%unit_cell_val = allegro_source%unit_cell_val
+
+   END SUBROUTINE pair_potential_allegro_copy
+
+! **************************************************************************************************
+!> \brief Creates the ALLEGRO potential type
+!> \param allegro ...
+!> \author Teodoro Laino [teo] 11.2005
+! **************************************************************************************************
+   SUBROUTINE pair_potential_allegro_clean(allegro)
+      TYPE(allegro_pot_type), POINTER                    :: allegro
+
+      IF (.NOT. ASSOCIATED(allegro)) RETURN
+      allegro%allegro_file_name = 'NULL'
+      allegro%allegro_version = 'NULL'
+      allegro%rcutsq = 0.0_dp
+      allegro%unit_energy_val = 0.0_dp
+      allegro%unit_coords_val = 0.0_dp
+      allegro%unit_forces_val = 0.0_dp
+      allegro%unit_cell_val = 0.0_dp
+      allegro%unit_coords = 'NULL'
+      allegro%unit_forces = 'NULL'
+      allegro%unit_energy = 'NULL'
+      allegro%unit_cell = 'NULL'
+
+   END SUBROUTINE pair_potential_allegro_clean
+
+! **************************************************************************************************
+!> \brief Destroys the ALLEGRO potential type
+!> \param allegro ...
+!> \author Teodoro Laino [teo] 11.2005
+! **************************************************************************************************
+   SUBROUTINE pair_potential_allegro_release(allegro)
+      TYPE(allegro_pot_type), POINTER                    :: allegro
+
+      IF (ASSOCIATED(allegro)) THEN
+         DEALLOCATE (allegro)
+      END IF
+   END SUBROUTINE pair_potential_allegro_release
```

Comment:
Do you really need this function (especially if you have removed the create routine)?

Reply:
As for the create routine, I would like to keep it for consistency with previous code.